ush/python/pygfs/task/atmens_analysis.py

```diff
@@ -285,21 +235,38 @@ def jedi2fv3inc(self: Analysis) -> None:
         # Reference the python script which does the actual work
         incpy = os.path.join(self.task_config.HOMEgfs, 'ush/jediinc2fv3.py')
 
+        # create template dictionaries
+        template_inc = self.task_config.COM_ATMOS_ANALYSIS_TMPL
+        tmpl_inc_dict = {
+            'ROTDIR': self.task_config.ROTDIR,
+            'RUN': self.runtime_config.RUN,
+            'YMD': to_YMD(self.task_config.current_cycle),
+            'HH': self.task_config.current_cycle.strftime('%H')
+        }
+
+        template_ges = self.task_config.COM_ATMOS_HISTORY_TMPL
+        tmpl_ges_dict = {
+            'ROTDIR': self.task_config.ROTDIR,
+            'RUN': self.runtime_config.RUN,
+            'YMD': to_YMD(self.task_config.previous_cycle),
+            'HH': self.task_config.previous_cycle.strftime('%H')
+        }
+
+        # loop over ensemble members
         for imem in range(1, self.task_config.NMEM_ENKF + 1):
             memchar = f"mem{imem:03d}"
 
             # make output directory for member increment
-            incdir = [
-                os.path.join(self.task_config.COMOUT, memchar, 'atmos')
-            ]
-            FileHandler({'mkdir': incdir}).sync()
+            tmpl_inc_dict['MEMDIR'] = memchar
+            incdir = Template.substitute_structure(template_inc, TemplateConstants.DOLLAR_CURLY_BRACE, tmpl_inc_dict.get)
+            FileHandler({'mkdir': [incdir]}).sync()
```

Comment:
We should try to collect all the actions (in this case mkdir for the different members) into a list and then call `FileHandler` once, instead of calling it in a loop over the ensemble members.
`FileHandler` will have the ability to handle data asynchronously.

Should we pull this out at the top of `initialize` where `DATA/bkg/memXXX` and `DATA/anl/memXXX` are being created?

Reply:
Good point @aerorahul .    Suggestion committed at [`292acce`](https://github.com/RussTreadon-NOAA/global-workflow/commit/292accebbb14dace68286c00d35dfe95e3fbdd57).